### PR TITLE
Allow internal input to inherit form-field props

### PR DESCRIPTION
**Pull request by pan-kot**
### Description

A second attempt for: https://github.com/cloudscape-design/components/pull/396

The new internal input behaviour is now under flag to not cause regression in select/multiselect.

### How has this been tested?

Added new tests.

### Documentation changes

[*Do the changes include any API documentation changes?*]
- [ ] _Yes, this change contains documentation changes._
- [x] _No._

### Related Links

* Ticket: AWSUI-19215

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- [ ] _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](CONTRIBUTING.md#public-apis)._
- [ ] _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](CONTRIBUTING.md#browsers-support)._
- [ ] _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- [ ] _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- [ ] _Changes are covered with new/existing unit tests?_
- [ ] _Changes are covered with new/existing integration tests?_
</details>

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/input/internal.tsx [hunk 1]
```diff
@@ -35,6 +35,8 @@ export interface InternalInputProps
 
   __onDelayedInput?: NonCancelableEventHandler<BaseChangeDetail>;
   __onBlurWithDetail?: NonCancelableEventHandler<{ relatedTarget: Node | null }>;
+
+  __useFormField?: boolean;
```
Comment (just-boris) [inline src/input/internal.tsx:39]: React naming suggests to avoid using `use-` for everything, which is not a hook. This one could be `inheritFormFieldLabels`

File: src/date-range-picker/__tests__/date-range-picker-relative.test.tsx [hunk 2]
```diff
@@ -198,7 +230,7 @@ describe('Date range picker', () => {
         relativeOptions: [],
         dateOnly: true,
       });
-      act(() => wrapper.findLabel().click());
+      act(() => wrapper.findTrigger().click());
```
Comment (just-boris) [inline src/date-range-picker/__tests__/date-range-picker-relative.test.tsx:233]: Why is this change needed? Also, does it affect customer tests if they use the same util?

**Review by YueyingLu — APPROVED** [review]

File: src/input/internal.tsx (hunk 1 above)
Comment (pan-kot) [reply src/input/internal.tsx:39]: Love that!

File: src/date-range-picker/__tests__/date-range-picker-relative.test.tsx (hunk 2 above)
Comment (pan-kot) [reply src/date-range-picker/__tests__/date-range-picker-relative.test.tsx:233]: These two are the same but `findLabel` is deprecated.

Comment (just-boris) [reply src/date-range-picker/__tests__/date-range-picker-relative.test.tsx:233]: Bonus points for moving this to a separate PR, but I can accept the current one too

Comment (pan-kot) [reply src/date-range-picker/__tests__/date-range-picker-relative.test.tsx:233]: Here is findLabel implementation:

```
findLabel(): ElementWrapper {
    return this.findTrigger();
}
```

Comment (just-boris) [reply src/date-range-picker/__tests__/date-range-picker-relative.test.tsx:233]: The PR title says
> Allow internal input to inherit form-field props

This change is unrelevant to the title. You can submit refactoring of these files in a separate PR

**Review by just-boris — APPROVED** [review]
